session/create_consumer.go

```diff
@@ -70,14 +80,33 @@ func (consumer *createConsumer) Consume(requestPtr interface{}) (response interf
 				destroyCallback()
 			}()
 		}
-		return responseWithSession(sessionInstance, config, nil), nil
+		return responseWithSession(sessionInstance, config, consumer.loadPaymentInfo(issuerID)), nil
 	case ErrorInvalidProposal:
 		return responseInvalidProposal, nil
 	default:
 		return responseInternalError, nil
 	}
 }
 
+func (consumer *createConsumer) loadPaymentInfo(issuerID identity.Identity) *PaymentInfo {
+	sp, err := consumer.promiseLoader.GetLastPromise(issuerID)
+	if err != nil {
+		log.Trace(consumerLogPrefix, "could not load promise info, defaulting to nil payment info", err)
+		return nil
```

Comment:
Here we probably should return empty, but initialised zero `PaymentInfo`. Further logic would not have to deal with nil pointer.